### PR TITLE
Barista/generate homepage

### LINKED ISSUE #136: Barista: create index-page-builder
Currently the Barista index page is hardcoded (at least most parts of it). Fetch the content from our CMS and generate the JSON-output for the index page.

### PR DESCRIPTION
**Pull request by lara-aigmueller**
### <strong>Pull Request</strong>

fixes #136 
blocked by #199 

<hr>
Hi, thank you for contributing to the Dynatrace Angular Component library (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please fill out the checkbox below: <br>

#### Type of PR

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or change that would cause existing functionality to
      not work as expected)
- [ ] Documentation update (changes to documentation)
- [x] Other (if none of the above apply)

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: apps/barista/src/layout/smalltile/smalltile.ts [hunk 1]
```diff
@@ -23,9 +23,36 @@ import { BaIndexPageItem } from '../../shared/page-contents';
...
   host: {
+    '[class]': '"ba-theme-" + _getTheme()',
     '[class.ba-smalltile-link-wrapper]': 'data',
   },
 })
 export class BaSmallTile {
   @Input() data: BaIndexPageItem;
+
+  /** @internal Gets the tile's theme based on its category. */
+  _getTheme(): string {
```
Comment (tomheller) [inline apps/barista/src/layout/smalltile/smalltile.ts]: Is there any specific reasoning behind, why this one and the `_identifier` are not getters?

File: apps/barista/src/layout/smalltile/smalltile.ts [hunk 2]
```diff
@@ -23,9 +23,36 @@ import { BaIndexPageItem } from '../../shared/page-contents';
...
+          return 'royalblue';
+        case 'Tools':
+          return 'yellow';
+      }
+    }
+    return 'turquoise';
+  }
+
+  /** @internal Gets the tile's identifier based on the title. */
+  _identifier(): string {
```
Comment (tomheller) [inline apps/barista/src/layout/smalltile/smalltile.ts]: Please change it to a getter.

File: apps/barista/src/layout/smalltile/smalltile.html [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <a *ngIf="data" [href]="data.link" class="ba-smalltile-link">
   <div class="ba-smalltile-img">
-    <span>{{ data.identifier }}</span>
+    <span>{{ _identifier() }}</span>
```
Comment (tomheller) [inline apps/barista/src/layout/smalltile/smalltile.html]: If the `identifier` is a getter, please use 
```suggestion
    <span>{{ _identifier }}</span>
```

File: apps/barista/src/shared/page-contents.ts [hunk 4]
```diff
@@ -53,3 +54,10 @@ export interface BaIndexPageItem {
 export interface BaOverviewPageContents extends BaPageMetaBase {
   sections: BaOverviewPageSection[];
 }
+
+export interface StrapiCTA {
```
Comment (tomheller) [inline apps/barista/src/shared/page-contents.ts]: Please change to a prefixed interface. 
```suggestion
export interface BaStrapiCTA {
```

It also collides with the interface barista-definitions.

File: apps/barista/src/shared/page-contents.ts [hunk 5]
```diff
@@ -20,6 +20,7 @@ export interface BaIndexPageContents extends BaPageMetaBase {
   subtitle: string;
   mostordered: BaIndexPageItem[];
   gettingstarted: BaIndexPageLink[];
+  cta: StrapiCTA;
```
Comment (tomheller) [inline apps/barista/src/shared/page-contents.ts]:
```suggestion
  cta: BaStrapiCTA;
```

**Comment by lara-aigmueller** [comment]
> The conflicting interfaces of `StrapiCTA` and `BaStrapiCTA` might be a little confusing. Would it be possible to think about a better separation of typing objects that come from strapis api, and ones that we use in the builders?

My idea was to use `BaStrapi...` as prefix for objects coming from the CMS and `Ba...` for objects generated by the page-builders and used in Barista. But as there were some PRs open at the same time, all modifying the types, it's a bit of a mess now... I'll try to do an interface cleanup in the end in a separate PR.

File: tools/barista/src/utils/fetch-strapi-content.ts [hunk 6]
```diff
@@ -86,8 +88,8 @@ async function fetchContent(requestPath: string): Promise<any> {
       });
 
       res.on('end', () => {
-        const jsonRes = JSON.parse(fullData);
-        if (jsonRes.length > 0) {
+        if (fullData.length > 0) {
```
Comment (lukasholzer) [inline tools/barista/src/utils/fetch-strapi-content.ts]: NIT: fullData.length would checks automatically if the length is greater than 0 ;)

**Pull request merged**